### PR TITLE
Display: Support `preserve_partitioning` on SortExec physical plan.

### LINKED ISSUE #10138: [Display] show preserve_partitioning in sortExec
### Is your feature request related to a problem or challenge?

Now when print plan of one sql in datafusion:
```
SortExec: TopK(fetch=10000), expr=[NASME@0 ASC,VISITS@1 DESC NULLS LAST]
```
Its difficult to find out what the sort unit is : **sort in partition** or **sort in global**

### Describe the solution you'd like

display `preserve_partitioning` info in SortExec
like
 ```
SortExec: TopK(fetch=10000), expr=[NASME@0 ASC,VISITS@1 DESC NULLS LAST], preserve_partitioning=[true] 
```

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kavirajk**
## Which issue does this PR close?

Closes #10138

## Rationale for this change

`Display` of SortExec execution plan doesn't have visibility on `preserve_partitioning` field
```
SortExec: TopK(fetch=10000), expr=[NASME@0 ASC,VISITS@1 DESC NULLS LAST]
```
## What changes are included in this PR?

Only important change made in this PR is

`Display` of SortExec execution plan now support `preserve_partitioning` field

```
SortExec: TopK(fetch=10000), expr=[NASME@0 ASC,VISITS@1 DESC NULLS LAST], preserve_partitioning=[true] 
```

Everything else is either updating tests manually or via some script. Following are the steps I used to update tests automatically (thanks @alamb for the tip)

1. `cargo test --test sqllogictests -- --complete` - that updates all `.slt` files except ones from `tpch` benchmarks
2. Generate `tpch` benchmarks data using steps mentioned[ here](https://github.com/apache/datafusion/tree/main/datafusion/sqllogictest#running-tests-tpch).
3. `INCLUDE_TPCH=true cargo test --test sqllogictests -- --complete` - to update `tpch` benchmarks tests outputs.

## Are these changes tested?

Yes
## Are there any user-facing changes?

May be, if user try to format the Sort execution plan.

### CODE REVIEW

**Comment by alamb** [comment]
I think you can update the tests using completion mode: https://github.com/apache/datafusion/tree/main/datafusion/sqllogictest#updating-tests-completion-mode

That probably will get this PR mostly working

**Comment by kavirajk** [comment]
@alamb thanks for the suggestion 👍 . Quick follow up. That command you mentioned `sqllogictests --complete` only updates the `.slt` files that are part of `datafusion/sqllogictest` crate itself.

I don't see it updates any of the other test cases where we are asserting the physical plan string representation. That still needs manual updates correct (around 79 tests were failing)?. 

I started updating those, just wanted to check if there are any easy/automated way that I'm not aware of it :)

File: datafusion/physical-plan/src/sorts/sort.rs [hunk 1]
```diff
@@ -868,11 +868,12 @@ impl DisplayAs for SortExec {
         match t {
             DisplayFormatType::Default | DisplayFormatType::Verbose => {
                 let expr = PhysicalSortExpr::format_list(&self.expr);
+                let preserve_partioning = self.preserve_partitioning;
```
Comment (kavirajk) [inline datafusion/physical-plan/src/sorts/sort.rs:871]: NOTE for reviewers: These changes in `sort.rs` is the only real change for this PR.

Everything else is just update tests either manually or automatically (see PR description for steps I used to update those)

**Comment by alamb** [comment]
I took the liberty of merging up from main to this branch to ensure we didn't have any logical conflicts (e.g. newly added explain plans for example). I think we can merge it once CI passes

**Pull request merged**